tests/smoke-tests/test-block-finalized.ts

```diff
@@ -0,0 +1,83 @@
+      const block = await context.polkadotApi.rpc.chain.getBlock(head);
+      const diff = Date.now() - getBlockTime(block);
+      debug(`Last finalized block was ${diff / 1000} seconds ago`);
+      expect(diff).to.be.lessThanOrEqual(10 * 60 * 1000); // 10 minutes in milliseconds
+    });
+
+    // TODO: Coordinate with Ops to make sure ETH RPC url is propagated before enabling this test
+    it("should have a recently finalized eth block", async function () {
+      const specVersion = context.polkadotApi.consts.system.version.specVersion.toNumber();
+      if (specVersion < 1900) {
+        debug(`ChainSpec ${specVersion} does not support Finalized BlockTag, skipping test`);
+        this.skip();
+      }
+      const timestamp = (await context.ethers.getBlock("finalized")).timestamp;
+      const diff = Date.now() - timestamp * 1000;
+      debug(`Last finalized block was ${diff / 1000} seconds ago`);
+      expect(diff).to.be.lessThanOrEqual(10 * 60 * 1000);
+    });
+
+    it("should have only finalized blocks in the past two hours.", async function () {
+      this.slow(10000);
+
+      const finalHash = await context.polkadotApi.rpc.chain.getFinalizedHead();
+      const signedBlock = await context.polkadotApi.rpc.chain.getBlock(finalHash);
+      const lastBlockNumber = signedBlock.block.header.number.toNumber();
+      const lastBlockTime = getBlockTime(signedBlock);
+      const limit = pLimit(5);
+
+      // Target time here is set to be 2 hours, possible parameterize this in future
+      const firstBlockTime = lastBlockTime - 2 * 60 * 60 * 1000;
+
+      const fetchBlockTime = async (blockNum) => {
+        const hash = await context.polkadotApi.rpc.chain.getBlockHash(blockNum);
+        const block = await context.polkadotApi.rpc.chain.getBlock(hash);
+        return getBlockTime(block);
+      };
+
+      const fetchHistoricBlockNum = async (blockNumber, targetTime) => {
+        return fetchBlockTime(blockNumber).then(async (time) => {
+          if (time < targetTime) {
+            return blockNumber;
+          } else {
+            return fetchHistoricBlockNum(
+              (blockNumber -= Math.ceil((time - targetTime) / 30_000)),
+              targetTime
+            );
+          }
+        });
+      };
+
+      const firstBlockNumber = await fetchHistoricBlockNum(lastBlockNumber, firstBlockTime);
+
+      debug(`Checking if blocks #${firstBlockNumber} - #${lastBlockNumber} are finalized.`);
+
+      const promises = (() => {
+        const length = lastBlockNumber - firstBlockNumber;
+        return Array.from({ length }, (_, i) => firstBlockNumber + i);
+      })().map((num) => limit(() => checkBlockFinalized(context.polkadotApi, num)));
+
+      const results = await Promise.all(promises);
+      results.forEach((item) => {
+        if (!item.finalized) debug(`Historic block #${item.number} is unfinalized!`);
+      });
+      expect(results.every((item) => item.finalized)).to.be.true;
```

Comment:
I'd suggest to put this expectation in the `forEach` above and print the block number that caused it to fail (fail fast). Or alternatively filter _all_ the blocks that are not finalized, and print it within this expect. Maybe printing all that failed is even better.

Reply:
done